### PR TITLE
Fix timezones uses

### PR DESCRIPTION
**Pull request by fedescarpa**
## :dart: Goal
Setting up TimeZone on every request to show times in current organization timezone

## :warning: Dependencies
https://github.com/mumuki/mumuki-domain/pull/219

## :back: Backwards compatibility
100%

### CODE REVIEW

**Review by julian-berbel — COMMENTED** [review]
We also leave_organization after request, I wonder if we should do that for `Time.zone` too?

File: app/views/exam_registrations/show.html.erb [hunk 1]
```diff
@@ -12,25 +12,27 @@
...
-    <%= form_for @authorization_request, html: {class: 'mu-form'} do |f| %>
-      <%= f.hidden_field :exam_registration_id, value: @registration.id %>
-      <%= f.label :exam_id, t(:exam_registration_choose_exam) %>
-      <% @registration.exams.each do |exam| %>
-        <div class="form-check">
-          <%= f.radio_button(:exam_id, exam.id, id: exam.id, required: true, class: 'form-check-input mu-read-only-input',
-                             checked: @authorization_request.exam_id == exam.id) %>
-          <%= label_tag exam.id, l(exam.start_time.in_time_zone(-3), format: :long), class: 'form-check-label' %>
-        </div>
+    <% unless @registration.end_time.past? %>
```
Comment (julian-berbel) [inline app/views/exam_registrations/show.html.erb:23]: is this validated in controller too?

Comment (fedescarpa) [reply app/views/exam_registrations/show.html.erb:23]: Now it is

File: app/views/exam_registrations/show.html.erb [hunk 2]
```diff
@@ -12,25 +12,27 @@
...
+      <%= form_for @authorization_request, html: {class: 'mu-form'} do |f| %>
+        <%= f.hidden_field :exam_registration_id, value: @registration.id %>
+        <%= f.label :exam_id, t(:exam_registration_choose_exam) %>
+        <% @registration.exams.each do |exam| %>
+          <div class="form-check">
+            <%= f.radio_button(:exam_id, exam.id, id: exam.id, required: true, class: 'form-check-input mu-read-only-input',
+                               checked: @authorization_request.exam_id == exam.id) %>
```
Comment (flbulgarelli) [inline app/views/exam_registrations/show.html.erb:30]: You can check this as `exam == exam`

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -143,6 +145,14 @@ def set_locale!
     I18n.locale = Organization.current.locale
   end
 
+  def set_time_zone!
+    Time.zone = Organization.current.time_zone
+  end
+
+  def unset_time_zone!
```
Comment (flbulgarelli) [inline app/controllers/application_controller.rb:152]: Why do we need to manually unset the time zone? I think we are not doing that with `current_user` and `Organization.current`, are we?

Comment (julian-berbel) [reply app/controllers/application_controller.rb:152]: > and `Organization.current`, are we?

we do, actually!
https://github.com/mumuki/mumuki-laboratory/blob/a41ffc14f7b45b5ed7c0578176f3b4caaacb9a74/app/controllers/application_controller.rb#L33
https://github.com/mumuki/mumuki-laboratory/blob/a41ffc14f7b45b5ed7c0578176f3b4caaacb9a74/app/controllers/application_controller.rb#L160-L162

as for current_user, not 100% sure, but I thought it doesn't survive the request as it's not saved as a thread variable?

Comment (julian-berbel) [reply app/controllers/application_controller.rb:152]: > as for current_user, not 100% sure, but I thought it doesn't survive the request as it's not saved as a thread variable?

https://stackoverflow.com/a/53671186

**Pull request merged**